### ISSUE TITLE
Please consider allowing combining --auto-correct with --only option, to check for just one problem domain (and have rubocop autocorrect only this problem, but not other problems, per given run)

### ISSUE THREAD

**Issue by shevegen** [bbatsov/rubocop#2190]
Hello.

I have recently discovered the --auto-correct option.

Allow me to say that this is an awesome option - I have used
it on some very old ruby files written by others, and it has
corrected about ~90% of the problems there automatically.

Since the code is very old though, I would like to go through
the options one step after the other.

I have lately discovered this option:

    rubocop --only EndAlignment

This is nice, rubocop tells me about wrong "end" statements.

I have an example file:

    class One

    class Two

    end; end

This class is nested into the other class. The two end statements
appear on the same line. Rubocop dislikes this according to the
specification/style guide.

All fine so far.

But now I would like of rubocop to autocorrect this, but only
this problem in the file.

So I was thinking I could combine it:

  rubocop --auto-correct --only EndAlignment test.rb

But it seems as if rubocop will ignore the --auto-correct
statement there. I am not sure why, because if I omit
--only then --auto-correct works fine.

Would it be possible to allow the above?

It should not be a huge change because there only would have
to be a flag that keeps track of the --only option used
there.

The reason why I would like this is because I would systematically
test one option after the other, and eliminate errors while
also inspecting the files (before, and then after the change).

If I --auto-correct all errors at the same time, without an
adapted rubocop.yml then this becomes quite overwhelming,
hence why I thought about the --only approach there.

Thank you for reading!

**Comment by bbatsov** [comment]
Hmm, I was under the impression that we made `--auto-correct` and `--only` play nice together. @jonas054 I guess I was wrong, right?

**Comment by jonas054** [comment]
We might have a problem here, but it's not related to the `--only` option.

> But it seems as if rubocop will ignore the --auto-correct
> statement there. I am not sure why, because if I omit
> --only then --auto-correct works fine.

Really? If I run without the `--only` option on your example I get
```ruby
class One
  class Two
  end; end
```
The indentation of the first `end` is the work of the `Style/IndentationWidth`. There's no cop that will complain about or fix the `;` in this code, and that's a problem IMO.

If we configure with
```yaml
Lint/EndAlignment:
  AutoCorrect: true
```
then `EndAlignment` will auto-correct misaligned `end`s, but only if they are at the beginning of a line.

**Comment by jefmathiot** [comment]
@bbatsov @jonas054 default configuration for `Lint/EndAlignment` and `Lint/DefEndAlignment` sets the `AutoCorrect` to `false`. Is there any particular reason these two cops disable autocorrect by default ? https://github.com/bbatsov/rubocop/blob/master/config/default.yml#L823

**Comment by bbatsov** [comment]
Yes. We have a rule that lint problems should not be auto-corrected by default, as they usually involve some subtleties.

**Comment by jefmathiot** [comment]
Thanks @bbatsov . These particular cops could be autocorrected by default, but disabled to ensure consistency across all lint cops, am I right ?

**Comment by jonas054** [comment]
No, there a specific reason for disabling this cop, not only that it belongs to the Lint family. See my answer in #2125.

**Comment by jefmathiot** [comment]
Makes sense, thanks a lot to both of you.

**Comment by alexdowad** [comment]
Looks like there is no issue here. Please close.

**Comment by paulslaby** [comment]
Hello guys, I have the same problem as the the opener had. 

I have a project I am trying to get in shape. I am dealing with it cop by cop. I just ran into situation, that i'd like to run `--auto-correct` and `--only`. but no luck: 

```> rubocop -a --only Style/NumericPredicate
403 files inspected, 75 offenses detected
> rubocop --auto-correct --only Style/NumericPredicate 
403 files inspected, 75 offenses detected
> rubocop --only Style/NumericPredicate --auto-correct
403 files inspected, 75 offenses detected
> rubocop -v
0.59.0```

**Comment by paulslaby** [comment]
I found the problem: You must configure Style/NumericPredicate to support `--auto-correct` with 
```
Style/NumericPredicate: 
  AutoCorrect: true
```

**Comment by WvanLelyveld** [comment]
`--auto-correct-all` is deprecated now, to `--autocorrect-all`.

So the new vresion would be:
`rubocop --autocorrect-all --only Style/TrailingCommaInHashLiteral `

**Comment by wyardley** [comment]
I get an error, for example with `RedundantCopDisableDirective`:
```
% bundle exec rubocop --autocorrect-all --only Lint/RedundantCopDisableDirective
Lint/RedundantCopDisableDirective cannot be used with --only.
```

**Comment by jonas054** [comment]
Yes, the `Lint/RedundantCopDisableDirective` cop is special. It doesn't work with `--only` for specific technical reasons. See
[this commit](https://github.com/rubocop/rubocop/pull/1896/commits/654591b9585fa07e5591e84b891395895bec9125). Note that the cop was called `UnneededDisable` back then.
